resources/assets/js/views/MenuEditor.vue

```diff
@@ -0,0 +1,400 @@
+import SitePageLinks from 'components/menu-editor/SitePageLinks';
+import { win, readingSpeedFromString, prettyDate } from 'classes/helpers';
+import Config from 'classes/Config';
+
+/* global setInterval, clearInterval */
+
+export default {
+	name: 'menu-editor',
+
+	components: {
+		Draggable,
+		Icon,
+		ScrollInput,
+		SitePageLinks,
+		MenuItemField
+	},
+
+	created() {
+		this.fetchSiteData();
+
+		this.validator = new Schema({
+			text: [
+				{
+					type: 'string',
+					required: true,
+					message: 'Link text can\'t be  empty.'
+				},
+				{
+					max: 100,
+					message: 'Link text must be less than 100 characters.'
+				}
+			],
+			url: [
+				{
+					required: true,
+					message: 'Location can\'t be  empty.'
+				},
+				{
+					type: 'url',
+					message: 'Location must be a valid URL.'
+				}
+			]
+		});
+
+		this.updateTimeElapsedTimer = setInterval(
+			this.updateTimeElapsedSincePublish,
+			5000
+		);
+	},
+
+	beforeRouteLeave(to, from, next) {
+		if(this.isUnsaved) {
+			this.$confirm(
+				'Are you sure you want to leave?',
+				'There are unsaved changes',
+				{
+					confirmButtonText: 'OK',
+					cancelButtonText: 'Cancel',
+					type: 'warning'
+				}
+			).then(() => {
+				next();
+			}).catch(() => {
+				next(false);
+			});
+		}
+		else {
+			next();
+		}
+	},
+
+	beforeDestroy() {
+		clearInterval(this.updateTimeElapsedTimer);
+	},
+
+	data() {
+		return {
+			site: {
+				firstPageId: 0,
+				title: '',
+				path: ''
+			},
+
+			menu: [],
+			errors: [],
+			sitePages: [],
+
+			// serialised version of the menu, to test equality
+			// with current menu for isUnsaved computed property
+			initialMenu: null,
+			// serialised version of the published menu, to test equality
+			// with current menu for status computed property
+			publishedMenu: null,
+			// raw published date as a string
+			lastPublishedDate: null,
+			// the time elapsed since ourlast publish, in a pretty format
+			timeElapsedSincePublish: null
+		};
+	},
+
+	computed: {
+		status() {
+			return (
+				// if unsaved but is the same as published menu, treat as draft
+				JSON.stringify(this.menu) === this.publishedMenu &&
+				!this.isUnsaved ?
+					'published' : 'draft'
+			);
+		},
+
+		isUnsaved() {
+			return JSON.stringify(this.menu) !== this.initialMenu;
+		},
+
+		datePublished() {
+			const date = new Date(this.lastPublishedDate);
+			return `${date.toDateString()} at ${date.getHours()}:${date.getMinutes()}`
+		}
+	},
+
+	methods: {
+
+		fetchSiteData() {
+			this.$api
+				.get(`sites/${this.$route.params.site_id}?include=pages`)
+				.then(({ data: json }) => {
+					const publishedMenu = (
+						json.data.options['menu_published'] ||
+						{ links: null, last_published: null }
+					);
+
+					this.site = {
+						firstPageId: json.data.pages[0].id,
+						title: json.data.name,
+						// TODO: don't hardcode HTTPS
+						path: 'https://' + json.data.host + json.data.path,
```

Comment:
maybe this could be '//' rather than 'https://'?

Reply:
We'd have to customise the validation, but it's possible. It might cause confusion to someone who's never seen a URL written like that, such as a publisher. Something to think about though.

Reply:
Yeah, we were thinking the same thing too. Maybe we could hide that part of the URL? 
You're right, something to think about for the future. Let's see how people end up using this.